aspnet/fundamentals/servers.rst

```diff
@@ -1,8 +1,259 @@
+
+ASP.NET 5 ships with support for 3 different servers:
+
+- Microsoft.AspNet.Server.IIS
+- Microsoft.AspNet.Server.WebListener (WebListener)
+- Microsoft.AspNet.Server.Kestrel (Kestrel)
+
+ASP.NET 5 does not directly listen for requests, but instead relies on the HTTP server implementation to surface the request to the application as a set of `feature interfaces`_ composed into an HttpContext. Both Helios and WebListener are Windows-only; Kestrel is designed to run cross-platform. You can configure your application to be hosted by any or all of these servers by specifying commands in your ``project.json`` file. You can even specify an application entry point for your application, and run it as an executable (using ``dnx . run``) rather than hosting it in a separate process.
+
+The default web host for ASP.NET applications developed using Visual Studio 2015 is IIS / IIS Express. The "Microsoft.AspNet.Server.IIS" dependency is included in ``project.json`` by default, even with the Empty web site template. Visual Studio provides support for multiple profiles, associated with IIS Express and any other ``commands`` defined in ``project.json``. You can manage these profiles and their settings in the Debug tab of your web application project's Properties menu.
+
+.. image:: /fundamentals/servers/_static/serverdemo-properties.png
+
+.. note:: IIS doesn't support commands; Visual Studio launches IIS Express and loads your application into it when you choose its profile.
+
+I've configured the sample project for this article to support each of the different server options in its ``project.json`` file:
+
+.. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/project.json
+	:lines: 1-17
+	:emphasize-lines: 12-14
+	:linenos:
+	:language: javascript
+	:caption: project.json (truncated)
+
+The ``run`` command will execute the application via its ``void main()`` method defined in ``program.cs``. In this case, this has been set up to configure and start an instance of ``WebListener``. This is not a typical means of launching a server, but is shown to demonstrate the possibility (the `Music Store sample application <https://github.com/aspnet/MusicStore>`_ also demonstrates this option).
+
+.. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/program.cs
+	:linenos:
+	:emphasize-lines: 27-35
+	:language: c#
+	:caption: program.cs
+
+
+Supported Features by Server
+----------------------------
+
+ASP.NET defines a number of :doc:`request-features` which may be supported on different server implementations. The following table lists the different features and the servers supporting them.
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Feature
+	  - IIS
+	  - WebListener
+	  - Kestrel
+	* - IHttpRequestFeature
+	  - Yes
+	  - Yes
+	  - Yes
+	* - IHttpResponseFeature
+	  - Yes
+	  - Yes
+	  - Yes
+	* - IHttpAuthenticationFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpUpgradeFeature
+	  - No
+	  - No
```

Comment:
Noted with
Yes (with limits)
Let me know if I can/should elaborate (and if so, what I should note, since I'm not familiar with its limits).